add undefined handling for no media

justfrankie opened the pull request:
was getting app-breaking errors accessing an undefined media post, optional parameters on these few places should bypass the break:

![image](https://github.com/hsu-isaac/fitconnect/assets/53983131/f1fdeb83-8a2d-47fe-b2e2-2eceb4db12e3)

justfrankie committed:
add undefined handling for no media

justfrankie commented on client/src/pages/Dashboard/MediaContainer.jsx:
```diff
@@ -3,7 +3,7 @@ import MediaRenderer from "./MediaRenderer";
 
 const MediaContainer = ({ mediaUrls }) => {
-  const numberOfMedia = mediaUrls.length;
+  const numberOfMedia = mediaUrls?.length;
```
fyi: https://www.freecodecamp.org/news/how-the-question-mark-works-in-javascript/

hsu-isaac replied:
This solution results with a different UI bug: 
<img width="1107" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/85271794/1f8b2dc5-cb4a-4d99-ac8d-8fff97c50457">

I have a solution in the branch that I'm working on now that handles this without rendering a large blank space for the media container, would you be ok with reverting this fix and pulling my branch into this after I push it to main and use [that solution](https://github.com/hsu-isaac/fitconnect/blob/add-edit-functionality/client/src/pages/Dashboard/Dashboard.jsx#L46C1-L49C19) instead?
It's essentially the same solution, I just have it one level higher for conditional container rendering

justfrankie replied:
yeah absolutely!! haha I wasn't sure what I was supposed to see/not see, I thought that was what the medias looked like since I was able to see some other medias! Let's ignore this and merge your fixes instead

justfrankie committed:
refactor buttons reusability and login page

justfrankie committed:
clean up login page refactor

justfrankie committed:
refactored sign-up page, added error messages

justfrankie commented:
<img width="794" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/0b438dcb-a9b4-4915-aca3-194001bb2792">
<img width="676" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/ccf179be-d110-4216-9bea-aa342ac651cd">

justfrankie commented on client/src/index.css:
```diff
@@ -1,7 +1,8 @@
 @import url("https://fonts.googleapis.com/css2?family=Nunito+Sans:wght@400;600;700&display=swap");
 
-.background-blue {
-  background-color: aqua;
```
removed since it wasn't being used

justfrankie commented on client/src/index.css:
```diff
@@ -41,38 +43,37 @@
   text-align: center;
 }
 
-.landing-button {
```
making buttons more resuable with just distinctive class names, like boostrap style

ideally, we could style commonly-used elements like buttons by adding class names "button-primary button-large button-hover-primary" etc

justfrankie committed:
remove unnecessary media query

justfrankie commented:
mobile responsiveness previews:

<img width="809" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/235fd9a8-0231-4bde-950e-a5ec55dc28e5">

<img width="842" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/93307973-8dfc-4196-b276-44032f466336">

hsu-isaac commented on client/src/pages/Login/LoginPage.jsx:
```diff
@@ -38,48 +44,72 @@ const LoginPage = () => {
         <form className={loginStyles["form"]}>
-          <label className={loginStyles["label"]}>
+          <div className={loginStyles["input-container"]}>
+            <label className={loginStyles["label"]} for="username">
```
I'm getting an error on my FE when I open this page:
<img width="1075" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/85271794/b168803e-fcbf-47db-b592-39c57ab48afe">

I'm not familiar with the 'for' syntax in label elements, would you mind explaining this to me?

justfrankie replied:
Huhh that's weird, let me get back to this and take a look at using "for".

I used the "for" atrtibute previously in labels so it attaches itself to an element:
https://www.w3schools.com/tags/tag_label.asp

I'll look into this :)

hsu-isaac commented on client/src/pages/Landing/Landing.css:
Was this the intended UI change?
<img width="1273" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/85271794/c300f8b1-ce32-4ff1-8f35-38441799990c">
(exposing my bookmarks lol)

All good if it is, I just thought the container looked a bit cramped/small and not fully centered on the page

justfrankie replied:
the container should be centered both from left, and the height was my choice, which we could definitely get back to that

hsu-isaac commented on client/src/components/layout/TopNavigation.jsx:
I think the CSS changes also caused the body to be created with no margin to the top of the page?
<img width="1280" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/85271794/e7470206-d2a7-4f6e-b4a7-13f272a60722">


Was this an intended design choice?

justfrankie replied:
oh that was somewhat intended, I must've removed the top/bottom margins to this class name, I can bring this back to what it was before!

The other margin removal I did was to the root container, since it did leave a little smalll gap at the top by default!

`* {
  margin: 0;
  padding: 0;
}`

hsu-isaac reviewed (changes requested):
Mostly looks good to me! I'm glad you're helping a lot with the styling aspect of things haha

Just found a couple bugs that I don't think were intended effects of the code change!

justfrankie committed:
Merge branch 'main' into undefined-properties-fix

justfrankie committed:
update label for attribute

justfrankie committed:
revert undefined/no workouts workaround

justfrankie committed:
center css for container and set restraint width for larger displays

justfrankie committed:
add padding parent-center class and some padding to nav bar

justfrankie committed:
fix small loader height issue

justfrankie commented:
fixes to comments earlier:
- Added more margin back to parent-center class for more space above the entire app container:
<img width="1511" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/0b3884d5-1027-4060-abbb-8fe3b54ede3a">

- Moved the app container "up" a little by setting `top: 15%` from 25% (can always fine tune)
<img width="1512" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/5805ed10-ebf1-4a8d-8538-822424459076">

- added max-width property so wide screens don't stretch out the main containers by percentage:
<img width="1512" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/7118bad6-1eca-41cd-a9c4-e1130c3f56f5">

- refactored `for` attribue in <label> since `for` keyword is reserved in jsx

-fixed height space between loading text and the spinny loading symbol
<img width="658" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/a7743b0e-2a1c-43f4-8186-c3f7e4f0efc1">

justfrankie commented on client/src/pages/Login/LoginPage.module.css:
```diff
@@ -1,8 +1,16 @@
+.input-container {
```
input bars preview:
<img width="696" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/53983131/7fea83b0-5a9f-4a54-8fd4-34b908ac43cd">

hsu-isaac replied:
<img width="115" alt="image" src="https://github.com/hsu-isaac/fitconnect/assets/85271794/cdfa5ece-fadf-4a84-9a20-78860945518a">


are these icons just a mac thing or something LOL i have no idea what they are and its not showing up when i run it locally - just wanted to make sure i wasnt missing anything

justfrankie committed:
remove creat react app traces

hsu-isaac reviewed (approved):
LGTM!
